pkg/lvm/controllerserver.go

```diff
@@ -285,3 +288,45 @@ func getControllerServiceCapabilities(cl []csi.ControllerServiceCapability_RPC_T
 
 	return csc
 }
+
+func (cs *controllerServer) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (*csi.GetCapacityResponse, error) {
+	if err := cs.validateControllerServiceRequest(csi.ControllerServiceCapability_RPC_GET_CAPACITY); err != nil {
+		klog.V(3).Infof("invalid get capacity req: %v", req)
+		return nil, err
+	}
+
+	nodeName := req.GetAccessibleTopology().GetSegments()[topologyKeyNode]
+	klog.Infof("Getting capacity for node: %s", nodeName)
+
+	lvmType := req.GetParameters()["type"]
+	switch lvmType {
+	case "linear", "mirror", "striped":
+		// These are supported lvm types
+	default:
+		return nil, status.Errorf(codes.Internal, "lvmType is incorrect: %s", lvmType)
+	}
+
+	va := volumeAction{
+		action:   "capacity",
+		name:     fmt.Sprintf("%s-%s", nodeName, lvmType),
+		nodeName: nodeName, lvmType: lvmType,
+		pullPolicy:       cs.pullPolicy,
+		provisionerImage: cs.provisionerImage,
+		kubeClient:       cs.kubeClient,
+		namespace:        cs.namespace,
+		vgName:           cs.vgName,
+		hostWritePath:    cs.hostWritePath,
+		devicesPattern:   cs.devicesPattern,
+	}
+	totalBytes, err := createCapacityPod(ctx, va)
+	if err != nil {
+		return nil, err
+	}
+
+	return &csi.GetCapacityResponse{
+		AvailableCapacity: totalBytes,
+		MaximumVolumeSize: wrapperspb.Int64(totalBytes),
```

Comment:
This cannot be same value?

Reply:
Looks like you are right. Can be looked up here: https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/csi-storage-capacity-v1/#CSIStorageCapacity